pandas/io/parsers/readers.py

```diff
@@ -83,13 +84,70 @@
         DtypeArg,
         DtypeBackend,
         FilePath,
+        HashableT,
         IndexLabel,
         ReadCsvBuffer,
         Self,
         StorageOptions,
         Unpack,
         UsecolsArgType,
     )
+
+    class _read_shared(TypedDict, Generic[HashableT], total=False):
```

Comment:
Moved in TYPE_CHECKING block - older python versions do not allow a TypedDict to inherit form non-TypedDict base classes (Generic)